mqtt-app-starters-common/src/main/java/org/springframework/cloud/stream/app/mqtt/MqttProperties.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.app.mqtt;
+
+import javax.validation.constraints.Size;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Generic mqtt connection properties.
+ *
+ * @author Janne Valkealahti
+ *
+ */
+@ConfigurationProperties("mqtt")
+public class MqttProperties {
+
+	/**
+	 * location of the mqtt broker(s) (comma-delimited list)
+	 */
+	private String[] url = new String[] { "tcp://localhost:1883" };
+
+	/**
+	 * the username to use when connecting to the broker
+	 */
+	private String username = "guest";
+
+	/**
+	 * the password to use when connecting to the broker
+	 */
+	private String password = "guest";
+
+	/**
+	 * whether the client and server should remember state across restarts and reconnects
+	 */
+	private boolean cleanSession = true;
+
+	/**
+	 * the connection timeout in seconds
+	 */
+	private int connectionTimeout = 30;
+
+	/**
+	 * the ping interval in seconds
+	 */
+	private int keepAliveInterval = 60;
+
+	/**
+	 * 'memory' or 'file'
+	 */
+	private String persistence = "memory";
+
+	/**
+	 * Persistence directory
+	 */
+	private String persistenceDirectory = "/tmp/paho";
+
+	@Size(min = 1)
```

Comment:
Adding validation annotations requires in the latest Boot to add `@Validated` to the class level: https://github.com/spring-cloud-stream-app-starters/app-starters-release/issues/82